lib/matplotlib/backends/backend_pgf.py

```diff
@@ -987,4 +989,177 @@ def _cleanup_all():
     LatexManager._cleanup_remaining_instances()
     TmpDirCleaner.cleanup_remaining_tmpdirs()
 
+
 atexit.register(_cleanup_all)
+
+
+class PdfPages(object):
+    """
+    A multi-page PDF file using the pgf backend
+
+    Examples
+    --------
+
+    >>> import matplotlib.pyplot as plt
+    >>> # Initialize:
+    >>> with PdfPages('foo.pdf') as pdf:
+    ...     # As many times as you like, create a figure fig and save it:
+    ...     fig = plt.figure()
+    ...     pdf.savefig(fig)
+    ...     # When no figure is specified the current figure is saved
+    ...     pdf.savefig()
+    """
+    __slots__ = (
+        'outputfile',
+        'keep_empty',
+        'tmpdir',
+        'base_name',
+        'fname_tex',
+        'fname_pdf',
+        '_n_figures',
+        '_file',
+    )
+
+    def __init__(self, filename, keep_empty=True, metadata=None):
+        """
+        Create a new PdfPages object.
+
+        Parameters
+        ----------
+
+        filename : str
+            Plots using :meth:`PdfPages.savefig` will be written to a file at
+            this location. Any older file with the same name is overwritten.
+        keep_empty : bool, optional
+            If set to False, then empty pdf files will be deleted automatically
+            when closed.
+        metadata : dictionary, optional
+            Information dictionary object (see PDF reference section 10.2.1
+            'Document Information Dictionary'), e.g.:
+            `{'Creator': 'My software', 'Author': 'Me',
+            'Title': 'Awesome fig'}`
+
+            The standard keys are `'Title'`, `'Author'`, `'Subject'`,
+            `'Keywords'`, `'Creator'`, `'Producer'`, `'CreationDate'`,
+            `'ModDate'`, and `'Trapped'`. Values have been predefined
+            for `'Creator'`, `'Producer'` and `'CreationDate'`. They
+            can be removed by setting them to `None`.
+        """
+        self.outputfile = filename
```

Comment:
all these attributes should be private unless you explicitly have a reason to make some public.

Reply:
sure